File: typescript-api/src/data/services/list-todo.ts
import { Todo } from "../../domain/entities";
import { ListTodoRepository } from "../repositories";

export class ListTodoService {
  constructor(
    private readonly repo: ListTodoRepository
  ) {}

  async list(userId: number): Promise<Todo[]> {
    return await this.repo.list(userId)
  }
}

File: rust-api/src/data/services/list_todo.rs
use crate::data::{ 
    repositories::ListTodoRepository,
    dto::TodoDto
};
use std::sync::Arc;

pub async fn list_todo_service(
    repo: Arc<impl ListTodoRepository>,
    user_id: i32,
) -> Result<Vec<TodoDto>, sqlx::Error> {
    repo.list(user_id).await
}
